applications/machine_learning/src/modules/led_state.c

```diff
@@ -31,12 +33,22 @@ BUILD_ASSERT(PREDICTION_STREAK_THRESH > 0);
 #define DEFAULT_EFFECT			(&ml_result_led_effects[0])
 
 static enum ml_state ml_state = ML_STATE_COUNT;
+static enum ei_data_forwarder_state forwarder_state = DISPLAY_DATA_FORWARDER ?
+	EI_DATA_FORWARDER_STATE_DISCONNECTED : EI_DATA_FORWARDER_STATE_TRANSMITTING;
+
 static const struct led_effect *blocking_led_effect;
 
 static const char *cur_label;
 static size_t prediction_streak;
 
 
+static bool is_led_effect_valid(const struct led_effect *le)
```

Comment:
Please leave the code as is but I want to add one comment regarding this part.

defensive programing is great but at some point you should consider following:
- adding protection code (any code) increase maintenance cost
- adding any code increases chances of adding bugs
- likelihood of some issues is next to none
- some issues are instantly visible and have otherwise no impact on the program

Reply:
I tried to introduce a complete validation for configuration defined by user in def file. Maybe it was not necessary in that case (since it's just related to LED effects).

Reply:
I will try to limit the validation in my next PRs